### DISCUSSION TITLE
Fluent theme does not fit vertical ProgressBar to Grid cell.

### DISCUSSION THREAD

**Discussion by stevensrmiller** [AvaloniaUI/Avalonia#16621] [Q&A] [answered]
When I put two ProgressBars into a Grid with one row and two columns, one bar with horizontal orientation and the other with vertical orientation, the horizontal bar is sized to fit the width of the cell. However, the vertical bar is set to its MinimumHeight value. The result looks like this:

![ProgFluent](https://github.com/user-attachments/assets/cec29ea9-013d-4a5b-b9af-144dbf0b1f8a)

Here's my code

	using Avalonia;
	using Avalonia.Controls;
	using Avalonia.Layout;

	class MainClass
	{
		static Window win;
		public static void Main(string[] args)
		{
			AppBuilder.Configure<Application>().UsePlatformDetect().Start(AppMain, args);
		}

		public static void AppMain(Application app, string[] args)
		{
			app.Styles.Add(new Avalonia.Themes.Fluent.FluentTheme());
	 
			win = new Window 
			{
				Width = 640,
				Height = 360,
			};

			var grid = new Grid
			{
				RowDefinitions = RowDefinitions.Parse("*"),
				ColumnDefinitions = ColumnDefinitions.Parse("*, *"),
				ShowGridLines = true,
			};

			win.Content = grid;

			var progressHorizontal = new ProgressBar
			{
				Height = 20,
				MinWidth = 10,
				Minimum = 0,
				Maximum = 100,
				Value = 50,
			};

			grid.Children.Add(progressHorizontal);

			var progressVertical = new ProgressBar
			{
				Width = 20,
				MinHeight = 10,
				Minimum = 0,
				Maximum = 100,
				Value = 50,
				Orientation = Orientation.Vertical,
			};

			progressVertical.SetValue(Grid.ColumnProperty, 1);

			grid.Children.Add(progressVertical);

			win.Show();
			app.Run(win);
		}
	}

If I change from Fluent theme to Simple theme, it works as expected:

![ProgSimple](https://github.com/user-attachments/assets/8a06021e-a739-42b2-8206-2c6ab19c2870)

Is this a bug or do I need to do something else to get both bars to fit to their grid cells in Fluent theme? Thanks!

[I'm new to Avalonia and this discussion group, so please offer any formatting or other advice you might have for a newcomer.]

**Comment by stevemonaco** [comment, marked as answer]
As Themes can define the entirety of a Control's visual appearance, they are not required to be consistent regarding stretching or spacing.

The vertical `ProgressBar` needs to be stretched in Fluent as the default is likely different. I've tested the following to work:

```cs
var progressVertical = new ProgressBar
{
    VerticalAlignment = VerticalAlignment.Stretch,
    Width = 20,
    MinHeight = 10,
    Minimum = 0,
    Maximum = 100,
    Value = 50,
    Orientation = Orientation.Vertical,
};
```

I'm not sure what's *expected* here from a MS Fluent perspective, so I can't comment if it's by-design or a bug that should be fixed.

**Reply by stevensrmiller** [reply]
Thanks, that does work. Interestingly, one does _not_ need to set the vertical alignment if the ProgressBar is replaced with a Slider.

Lots for me to learn, here. I appreciate the help.
